VERSION-RANGE-SPEC.rst

```diff
@@ -0,0 +1,989 @@
+in API with larger JSON documents.
+
+
+Why not use the OSV Ranges?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+See:
+
+- https://ossf.github.io/osv-schema/
+
+``vers`` and the OSSF OSV schema vulnerable ranges are equivalent and ``vers``
+provides a compact range notation while OSV provides more verbose JSON notation.
+
+``vers`` borrows the design from and was informed by the OSV schema spec and its
+authors.
+
+OSV uses a minimalist set of only three comparators:
+
+- "=" to enumerate versions,
+- ">=" for the version that introduced a vulnerability, and
+- "<"  for the version that fixed a vulnerability.
+
+OSV Ranges support neither ">" nor "!=" comparators making it difficult to
+express some ranges that must exclude a version. This may not be an issue for
+most vulnerable ranges yet:
+
+- this makes it difficult or impossible to precisely express certain dependency
+  and vulnerable ranges when a version must be excluded and the set of existing
+  versions is not yet known,
+
+- this make some ranges more verbose such as with the NVD CVE v5 API ranges
+  notation that can include their upper limit and would need two constraints.
+
+Another high level difference between the two specifications are the
+codes used to qualify a range package  "ecosystem" value that resembles closely
+the Package URL package "type" used in ``vers``. This spec will provide a strict
+mapping between the OSV ecosystem and the ``vers`` versioning schemes values.
+
+
+Why not use the NVD CVE v5 API Ranges?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+See:
+
+- https://github.com/CVEProject/cve-schema/blob/master/schema/v5.0/CVE_JSON_5.0_schema.json#L303
+- https://github.com/CVEProject/cve-schema/blob/master/schema/v5.0/CVE_JSON_5.0_schema.json#L123
+
+The version 5 of the NVD CVE JSON data format defines version ranges with a
+starting version, a versionType, and an upper limit for the version range as
+lessThan or lessThanOrEqual or as an enumeration of versions. The versionType
+and the package collectionURL possible values are only indicative and left out
+of this specification and both seem strictly equivalent to the Package URL
+"type" on the one hand and the ``vers`` versioning scheme on the other hand.
+
+The semantics and expressiveness of each range are similar and ``vers`` provides
+a compact notation rather than a more verbose JSON notation. ``vers`` supports
+strictly the conversion of any NVD v5 range to its notation and further
+provides a concrete list of well known versioning schemes. ``vers`` design was
+informed by the NVD CVE v5 API schema spec and its authors.
+
+When NVD v5 becomes active, this spec will provide a strict mapping between the
+NVD versionType and the ``vers`` versioning schemes values. Furthermore, this
+spec and the Package URL "types" should be updated accordingly to provide
+a mapping with the upcoming NVD collectionURL that will be effectively used.
+
+There is one issue with NVD v5: it introduces a new trailing "*" notation that
```

Comment:
@chandanbn @rsc is this right?

Reply:
@chandanbn @rsc  gentle ping wrt. the introduction of a "star" in the new CVE spec. What's the rationale and use case?